scipy/special/tests/test_mpmath.py

```diff
@@ -2239,8 +2239,9 @@ def mp_spherical_in(n, z):
 
     def test_spherical_kn(self):
         def mp_spherical_kn(n, z):
-            out = (mpmath.besselk(n + mpmath.mpf(1)/2, z) *
-                   mpmath.sqrt(mpmath.pi/(2*mpmath.mpmathify(z))))
+            arg = mpmath.mpmathify(z)
+            out = (mpmath.besselk(n + mpmath.mpf(1)/2, arg) /
+                   mpmath.sqrt(2*arg/mpmath.pi))
```

Comment:
One of those dark corners of floating point math that I hadn't encountered before:
```python3
import math
from mpmath import mp
x = -1 + 0j
((1/x)**0.5).imag  # -1.0
((1/x+0)**0.5).imag  # 1.0
((1/x)**mp.mpf(0.5)).imag  # mpf('1.0')
```

Apparently our complex implementation uses the negative (imaginary) square root.

Reply:
Cool, right? Check out http://port70.net/~nsz/articles/float/kahan_branch_cuts_complex_elementary_functions_1987.pdf if you’re interested